### PR TITLE
kblueberry ToDo List

### PR DESCRIPTION
**Pull request by kblueberry**

### CODE REVIEW

File: Khrystyna_Derenivska/src/app/new-item/new-item.component.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  styleUrls: ['./new-item.component.css']
+})
+export class NewItemComponent implements OnInit {
+  allowNewItem = false;
+  itemCreationStatus = 'No item was created';
+  serverName = 'Test task';
+  itemCreated = false;
+
+  constructor() {
+    setTimeout(() => {
```
Comment (mefissto) [inline Khrystyna_Derenivska/src/app/new-item/new-item.component.ts]: Clear the timeout

File: Khrystyna_Derenivska/src/app/new-item/new-item.component.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      this.allowNewItem = true;
+    }, 2000);
+  }
+
+  ngOnInit(): void {}
+
+  onCreateButtonClick() {
+    this.itemCreated = true;
+    this.itemCreationStatus = `Item was created. Name is ' + this.serverName'`;
```
Comment (mefissto) [inline Khrystyna_Derenivska/src/app/new-item/new-item.component.ts]: If you use a template string you can write something like that --> ``Item was created. Name is ${this.serverName}``

File: Khrystyna_Derenivska/src/app/new-item/new-item.component.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  }
+
+  onUpdateItemName(event: any) {
```
Comment (mefissto) [inline Khrystyna_Derenivska/src/app/new-item/new-item.component.ts]: Avoid the "any" type

File: Khrystyna_Derenivska/src/app/task-item/task-item.component.html [hunk 4]
```diff
@@ -0,0 +1,30 @@
+<div class="list__item" [ngClass]="{'task__done': task.isDone}">
+    <div *ngIf="isEditMode; then editTask else showTask"></div>
+    <ng-template #editTask>
+        <input type="text" id="edit-task-name" [(ngModel)]='editedTaskName'>
+    </ng-template>
+    <ng-template #showTask>
+        <div class="list__item_name" [ngClass]=" ">
+            <input type="checkbox" id="list__item-1_value" class="list__item_value" (change)="onDoneChange()"
+                [attr.disabled]="task.isDone ? '' : null">
```
Comment (mefissto) [inline Khrystyna_Derenivska/src/app/task-item/task-item.component.html]: Why do you use '' and null as value? If you need to set a disable attribute you can use only property "task.isDone" without a ternary operator.

File: Khrystyna_Derenivska/src/app/tasks-to-do-list/tasks-to-do-list.component.html [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<div class="tasks__container">
+    <h3 class="container-name">Tasks to do</h3>
+    <div class="tasks__list">
+        <app-task-item></app-task-item>
```
Comment (mefissto) [inline Khrystyna_Derenivska/src/app/tasks-to-do-list/tasks-to-do-list.component.html]: Make a list of tasks with ngFor directive

File: Khrystyna_Derenivska/src/app/new-item/new-item.component.ts (hunk 2 above)
Comment (kblueberry) [reply Khrystyna_Derenivska/src/app/new-item/new-item.component.ts]: Ok, thanks for the recommendation!

File: Khrystyna_Derenivska/src/app/app.component.html [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<!-- <app-new-item-add #appChild></app-new-item-add> -->
```
Comment (KolosUK) [inline Khrystyna_Derenivska/src/app/app.component.html]: Delete commented code before create a PR

File: Khrystyna_Derenivska/src/app/app.component.html [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+<div class="new-item-add">
+    <label for="new-task-name">Add new task</label>
+    <input type="text" id="add-task-name" [(ngModel)]='taskName'>
```
Comment (KolosUK) [inline Khrystyna_Derenivska/src/app/app.component.html]: Bad name for model

File: Khrystyna_Derenivska/src/app/app.component.html [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+    <button class="add-button" (click)="onCreateButtonClick()">Add</button>
+</div>
+
+<hr>
+<app-task-item *ngFor="let task of tasks" [task]='task' (deleteItem)="onDelete($event)"></app-task-item>
```
Comment (KolosUK) [inline Khrystyna_Derenivska/src/app/app.component.html]: Why you add 'on' to name of method?

**Review by KolosUK — COMMENTED** [review]
1) Delete  .spec.ts files

File: Khrystyna_Derenivska/src/app/app.component.html (hunk 6 above)
Comment (kblueberry) [reply Khrystyna_Derenivska/src/app/app.component.html]: Deleted!

File: Khrystyna_Derenivska/src/app/new-item/new-item.component.ts (hunk 1 above)
Comment (kblueberry) [reply Khrystyna_Derenivska/src/app/new-item/new-item.component.ts]: Ok, I'll remove it

File: Khrystyna_Derenivska/src/app/task-item/task-item.component.html (hunk 4 above)
Comment (kblueberry) [reply Khrystyna_Derenivska/src/app/task-item/task-item.component.html]: Good point.

File: Khrystyna_Derenivska/src/app/app.component.html (hunk 8 above)
Comment (kblueberry) [reply Khrystyna_Derenivska/src/app/app.component.html]: I've used to do that when I write event handler function

File: Khrystyna_Derenivska/src/app/app.component.html (hunk 6 above)
Comment (kblueberry) [reply Khrystyna_Derenivska/src/app/app.component.html]: Do I also need to delete tsconfig.spec.json file?

**Review by LTania — APPROVED** [review]

**Comment by kblueberry** [comment]
@imotality @sidgar21, @taraswithjazz  please, approve my changes in PR.
@mefissto, I can't re-request a review from you.

**Review by imotality — APPROVED** [review]

**Review by taraswithjazz — APPROVED** [review]

**Review by sidgar21 — APPROVED** [review]
Approve

**Review by mefissto — APPROVED** [review]

**Review by KolosUK — APPROVED** [review]

**Pull request merged**